Buildings/Controls/OBC/Utilities/PIDWithAutotuning/FirstOrderAMIGO.mo

```diff
@@ -190,11 +176,50 @@ protected
     "Check if the relay output is asymmetric"
     annotation (Placement(transformation(extent={{20,140},{40,160}})));
   Buildings.Controls.OBC.CDL.Reals.Sources.Constant con4(final k=1e-3)
-    "Threshold for checking the symmetricity of the relay output"
+    "Threshold for checking if the input is larger than 0"
     annotation (Placement(transformation(extent={{-20,110},{0,130}})));
   Buildings.Controls.OBC.CDL.Reals.MultiplyByParameter yRel(final k=yMax - yMin)
-    "Relay output multiplied by the possible range of the output "
+    "Relay output multiplied by the possible range of the output"
     annotation (Placement(transformation(extent={{40,110},{60,130}})));
+  Buildings.Controls.OBC.CDL.Logical.Nand nand
+    "Check if an autotuning is ongoing while a new autotuning request is received"
+    annotation (Placement(transformation(extent={{120,-72},{140,-52}})));
+  Buildings.Controls.OBC.CDL.Utilities.Assert assMes1(message="*** Warning: An autotuning is ongoing and the new autotuning request is ignored.")
+    "Warning message when an autotuning tuning 
+    is ongoing while a new autotuning request is received"
+    annotation (Placement(transformation(extent={{148,-72},{168,-52}})));
+  Buildings.Controls.OBC.CDL.Logical.Edge edgReq
+    "True only when a new request is received"
+    annotation (Placement(transformation(extent={{80,-80},{100,-60}})));
+  Buildings.Controls.OBC.CDL.Logical.TrueDelay tunStaDel(
+    final delayTime=0.001)
+    "A small time delay for the autotuning start time to avoid false alerts"
+    annotation (Placement(transformation(extent={{80,-40},{100,-20}})));
+  Buildings.Controls.OBC.CDL.Reals.AddParameter addPar(
+    final p=yMin)
+    "Sums the inputs"
+    annotation (Placement(transformation(extent={{80,110},{100,130}})));
+  Buildings.Controls.OBC.CDL.Discrete.TriggeredSampler sam_u_s(
+    final y_start=u_s_start)
+    "Recording the setpoint"
+    annotation (Placement(transformation(extent={{-100,240},{-80,220}})));
+  Buildings.Controls.OBC.CDL.Logical.Nand nand1
+    "Check if an autotuning is ongoing while the setpoint changes"
+    annotation (Placement(transformation(extent={{100,220},{120,240}})));
+  Buildings.Controls.OBC.CDL.Reals.Subtract sub3
+    "Change of the setpoint"
+    annotation (Placement(transformation(extent={{-38,220},{-18,240}})));
+  Buildings.Controls.OBC.CDL.Reals.Abs abs2
+    "Absolute value of the setpoint change"
+    annotation (Placement(transformation(extent={{0,220},{20,240}})));
+  Buildings.Controls.OBC.CDL.Reals.Greater gre1
```

Comment:
we should have a hysteresis, `h` should be greater than zero. You could set it  as like `h=setHys` and set the `setHys` to be a flexible value, e.g. `0.05*r`